### PR TITLE
installing dependencies when needed

### LINKED ISSUE #1738: installing dependencies when needed
# 🙋 Feature Request

### 😯 Current Behavior

pulling and checkout different code doesn't trigger anything

### 🤔 Expected Behavior

Change in dependencies should trigger a process which installs the relevant dependencies (both for internal usage and user's code)

### 💁 Possible Solution 

starting off by just running `npm install` in a git hook that checks for `package.json` file change. checkout #1737 

maybe later we can diff for changes and install just the relevant dependencies.

### PR DESCRIPTION
**Pull request by deanshub**
### 🔦 Summary

Fixes #1738 

when pulling changes or checking out a different branch often dependencies change.
This PR should handle changes and install dependencies when needed.

### CODE REVIEW

**Comment by amitdahan** [comment]
How about asking before running `npm install`? :-D

File: scripts/post-merge [hunk 1]
```diff
@@ -6,8 +6,22 @@
 
 changed_files="$(git diff-tree -r --name-only --no-commit-id ORIG_HEAD HEAD)"
 
+# Confirmation before installing
+confirm() {
+    # call with a prompt string or use a default
+    read -r -p "Some dependencies changed, should I run \`npm install\`? [y/N]" response
+    case "$response" in
+        [yY][eE][sS]|[yY])
```
Comment (amitdahan) [inline scripts/post-merge]: Eize Gever!!!

File: package.json [hunk 2]
```diff
@@ -41,7 +41,8 @@
   },
   "husky": {
     "hooks": {
-      "pre-commit": "lint-staged"
+      "pre-commit": "lint-staged",
+      "post-merge": "ts-node -T scripts/post-merge"
```
Comment (ranyitz) [inline package.json]: Just to make sure that we're not missing anything, would we have `ts-node` on CI agents when running post-merge hook?

Comment (deanshub) [reply package.json]: don't think we need it since CI just colnes and doesn't pull on existing branch (this hook shouldn't run)

File: yarn.lock [hunk 3]
```diff
@@ -4473,6 +4473,13 @@
   resolved "https://repo.dev.wixpress.com/artifactory/api/npm/npm-repos/@zeit/schemas/-/schemas-2.6.0.tgz#004e8e553b4cd53d538bd38eac7bcbf58a867fe3"
   integrity sha1-AE6OVTtM1T1Ti9OOrHvL9YqGf+M=
 
+"@zeit/webpack-asset-relocator-loader@^0.6.2":
```
Comment (ranyitz) [inline yarn.lock]: not related to this PR

File: yarn.lock [hunk 4]
```diff
@@ -4473,13 +4473,6 @@
   resolved "https://repo.dev.wixpress.com/artifactory/api/npm/npm-repos/@zeit/schemas/-/schemas-2.6.0.tgz#004e8e553b4cd53d538bd38eac7bcbf58a867fe3"
   integrity sha1-AE6OVTtM1T1Ti9OOrHvL9YqGf+M=
 
-"@zeit/webpack-asset-relocator-loader@^0.6.2":
```
Comment (ranyitz) [inline yarn.lock]: Why the `yarn.lock` needed a change?

**Pull request merged**